common/src/lib.rs

```diff
@@ -176,7 +176,14 @@ impl BlockHeader {
 
 #[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Clone)]
 pub struct MerkleProof {
-    // TODO
+    pub proof: Vec<MerkleTree>,
+}
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Eq, Clone)]
+pub enum MerkleTree {
+    // (pair_hash, is_right_node)
+    HasSibling(Hash256, bool),
+    HasNoSibling,
```

Comment:
Sorry for my miscommunication, but what I was think of is
`HasRightSibling` and `HasLeftSibling`. This will improve the overall readability .

Reply:
Understood. Fixed.